service/matching/tasklist/task_list_manager.go

```diff
@@ -796,31 +797,14 @@ func (c *taskListManagerImpl) getIsolationGroupForTask(ctx context.Context, task
 		}
 		group, err := c.partitioner.GetIsolationGroupByDomainID(ctx, taskInfo.DomainID, partitionConfig, pollerIsolationGroups)
 		if err != nil {
-			// For a sticky tasklist, return StickyUnavailableError to let it be added to the non-sticky tasklist.
-			if err == partition.ErrNoIsolationGroupsAvailable && c.taskListKind == types.TaskListKindSticky {
-				return "", _stickyPollerUnavailableError
-			}
 			// if we're unable to get the isolation group, log the error and fallback to no isolation
 			c.logger.Error("Failed to get isolation group from partition library", tag.WorkflowID(taskInfo.WorkflowID), tag.WorkflowRunID(taskInfo.RunID), tag.TaskID(taskInfo.TaskID), tag.Error(err))
-			return defaultTaskBufferIsolationGroup, nil
+			return defaultTaskBufferIsolationGroup, noIsolationTimeout, nil
 		}
 		c.logger.Debug("get isolation group", tag.PollerGroups(pollerIsolationGroups), tag.IsolationGroup(group), tag.PartitionConfig(partitionConfig))
-		// For a sticky tasklist, it is possible that when an isolation group is undrained, the tasks from one workflow is reassigned
```

Comment:
not sure I understand the thinking behind removing this?

Reply:
Sorry I should have left a comment explaining this.

This is actually all dead code after this change: https://github.com/cadence-workflow/cadence/commit/affb15bb3cb291944d4a493f7795ac0aa4227455 . Line 780 means that this condition is never true, same with the conditions I removed on lines 790 and 799.

Like Zijian points out though, we don't need zonal isolation for sticky task lists at all. We get it for free since it's isolated to the host. If we hit a sticky poller unavailable then it'll fall back to the normal task list, which will have zonal isolation, and then get assigned to a new host within that same zone (unless it ends up leaked). It would only get leaked currently if that zone is drained or we haven't seen a poller from that zone in the last 10 seconds.